runtime/libc/wasi_sdk_backing.c

```diff
@@ -339,7 +334,45 @@ i32 wasi_unstable_path_open(
     return WASI_ESUCCESS;
 }
 
-i32 wasi_unstable_fd_close(i32 fd) {
+/**
+ * @brief Used by a WASI module to copy the argument buffer into linear memory write an 
+ * indirect to the base offset of the buffer 
+ * 
+ * @param argv_p_offset 
+ * @param argv_buf_offset 
+ * @return i32 
+ */
+i32 wasi_snapshot_preview1_args_get(i32 argv_p_offset, i32 argv_buf_offset){
+    char *argv_buf = get_memory_ptr_for_runtime(argv_buf_offset, runtime_argv_buffer_len);
+    strncpy(argv_buf, runtime_argv_buffer, runtime_argv_buffer_len);
+
+    u32 *argv_p = (u32 *)get_memory_ptr_for_runtime(argv_p_offset, sizeof(i32));
+    *argv_p = argv_buf_offset;
```

Comment:
Strange API that passes in a chunk of memory and expects a pointer to it back. Seems odd.

https://docs.rs/wasi/0.9.0+wasi-snapshot-preview1/wasi/fn.args_get.html confirms your intuition. I might put `args_sizes_get` before this function to mimic the dependency.